relay-redis/src/pool.rs

```diff
@@ -186,7 +199,19 @@ impl Manager for CustomClusterManager {
             return Ok(());
         }
 
-        self.manager.recycle(conn, metrics).await
+        // Copied from deadpool_redis::cluster::Manager
+        let ping_number = self.ping_number.fetch_add(1, Ordering::Relaxed).to_string();
+        let n = redis::cmd("PING")
+            .arg(&ping_number)
+            .query_async::<String>(conn)
+            .await?;
+        if n == ping_number {
+            Ok(())
+        } else {
+            Err(deadpool::managed::RecycleError::message(
+                "Invalid PING response",
+            ))
```

Comment:
```suggestion
            Err(RecycleError::message("Invalid PING response"))
```

`RecycleError` is already imported.

Reply:
Good catch!